### PR TITLE
SE2/SE3/SO3 - consolidate EPS, add dtype-conditioned EPS and add float32 unit tests

### LINKED ISSUE #70: Add all the EPS to `constants.py`
Should this go to `constants.py`? Was this arbitrarily set or comes from another source?

_Originally posted by @mhmukadam in https://github.com/facebookresearch/theseus/pull/46#discussion_r795569723_

### PR DESCRIPTION
**Pull request by luisenp**
Closes #70.

### CODE REVIEW

**Review by fantaosha — COMMENTED** [review]
LGTM. Only some minor comments.

**Comment by mhmukadam** [comment]
Looks like int64 is being passed in one of the unit tests for the dtype to get EPS causing CI to fail.

File: theseus/geometry/se2.py [hunk 1]
```diff
@@ -229,7 +232,7 @@ def exp_map(
         cosine, sine = rotation.to_cos_sin()
 
         # Compute the approximations when theta is near to 0
-        small_theta = theta.abs() < SE2.SE2_EPS
+        small_theta = theta.abs() < theseus.constants._SE2_EPS[tangent_vector.dtype]
```
[the remark this answers is not in the document]
Comment (luisenp) [reply theseus/geometry/se2.py:235]: This is a static method.

File: theseus/geometry/se2.py [hunk 2]
```diff
@@ -166,7 +169,7 @@ def _log_map_impl(
         cosine, sine = rotation.to_cos_sin()
 
         # Compute the approximations when theta is near to 0
-        small_theta = theta.abs() < SE2.SE2_EPS
+        small_theta = theta.abs() < self._EPS
```
[the remark this answers is not in the document]
Comment (luisenp) [reply theseus/geometry/se2.py]: Changed.

**Review by mhmukadam — APPROVED** [review]

**Review by mhmukadam — APPROVED** [review]

**Pull request merged**